### DISCUSSION TITLE
Unable to syntax highlight rules in Webview

### DISCUSSION THREAD

**Discussion by bk201-** [microsoft/vscode-discussions#1713] [Extension Development QnA] [answered]
VSCode passes to Webview theme's variables (--vscode-<tokenname>). It allows to implement the same theme in webview as in VSCode. https://github.com/microsoft/vscode/blob/9e23739d0b9f3381d99be9bde649709d4329daa5/src/vs/workbench/contrib/webview/browser/themeing.ts#L52 

However, vscode doesn't pass rules for syntax highlight to implement syntax theme in Monaco.

Is it possible retrieve these rules directly from the code (extension side or webview side)?

**Comment by isidorn** [comment, marked as answer]
I think this is currently not supported by the extension API.

**Reply by bk201-** [reply]
Is it any chance to implement this? I'm sure not only my team use Monaco Editor in webviews in extensions, notebooks are not always appropriated for implementing different features.

**Reply by burkeholland** [reply]
@bk201- Could you please open this as a feature request on the VS Code repo? https://github.com/microsoft/vscode
